class/practice2.1/AdvancedHelloWorld.cpp

```diff
@@ -1,110 +1,111 @@
-        tokens.push_back(token);
-    }
-    return tokens;
+#include <map>
+#include <print>
+namespace {
+   constexpr const char* bread = "bread";
+   constexpr const char* deleteCommand = "delete";
 }
-int main(int argc, char* argv[])
-{
-    std::string argument2;
-        
-    std::string WordToFind;
-    if (argc < 2) {
-        std::cout << "Please provide a word to find." << std::endl;
-        return 0;
+class Database {
+
+public:
+    Database(const std::string& filePath) : databasePath(filePath) {
+         LoadData();
     }
-    WordToFind= argv[1];
-    if (argc>2)
-    {
-        argument2 = argv[2];
+    void clearDatabase() {
+        statsData.clear();
+        SaveData();
     }
-    if (WordToFind=="bread")
-    {
-        std::ofstream file("database.txt", std::ios::trunc); 
-        file.close();
-        std::cout << "all data was cleared" << std::endl;
-        return 0;
+
+    bool deleteUser(const std::string& nameToFind) {
+        auto nameIt = statsData.find(nameToFind);
+        if (nameIt != statsData.end()) {
+            statsData.erase(nameIt);
+            SaveData();
+            return true;
+        }
+        return false;
     }
-    std::ifstream file("database.txt");
-    std::string line;
-    int lineNumber = 0;
-    bool found = false;
-    std::string lineFound;
 
-    std::vector<std::string> lines;
- 
-    while (std::getline(file, line)) {
-        lines.push_back(line);
+    int UserIncrement(const std::string& nameToFind) {
+         statsData[nameToFind]++;
+         SaveData();
+         return statsData[nameToFind];
     }
-    file.close();
-    for (const std::string& line : lines) {
-        lineNumber++;
-        if (line != "") {
-            if (splitString(line, ' ')[0] == WordToFind)
-            {
-                found = true;
-                lineFound = line;
-                break;
-            }
-        }
+    bool checkName(const std::string& nameToFind){
+        return statsData.find(nameToFind)!=statsData.end();
     }
-    if (!found) {
-        if (argument2 == "delete") {
-            std::cout << "There is no such user " <<std::endl;
-            return 0;
+private:
+    std::map<std::string, int> statsData;
+    const std::string databasePath;
+    void LoadData() {
+        std::ifstream file(databasePath);
+        std::string name;
+        int count;
+        while (file >> name >> count) {
+            statsData [name] = count;
         }
-        std::cout << "Welcome " << WordToFind << std::endl;
-        std::ofstream fileEdit("database.txt", std::ios::app);
-        if (!fileEdit.is_open()) {
-            std::cerr << "Could not open the file for writing!" << std::endl;
-            return 1;
+        file.close();
+    }
+    void SaveData() {
+        std::ofstream file(databasePath, std::ios::trunc);
+        for (const auto& [name, count] : statsData ) {
+            file << name << " " << count << std::endl;
```

Comment:
Minor: you can use std::println to write formatted data to the file
`std::println(file, "{} {}", name, count);`